discretisedfield/field.py

```diff
@@ -2771,71 +2769,61 @@ def _writeovf(self, filename, representation='txt', extend_scalar=False):
                   f'ymax: {self.mesh.region.pmax[1]}',
                   f'zmax: {self.mesh.region.pmax[2]}',
                   f'valuedim: {write_dim}',
-                  'valuelabels: field_x field_y field_z',
-                  'valueunits: None None None',
+                  f'valuelabels: {valuelabels}',
+                  f'valueunits: {valueunits}',
                   '',
                   'End: Header',
                   '']
 
         if representation == 'bin4':
-            header.append('Begin: Data Binary 4')
-            footer = ['End: Data Binary 4',
-                      'End: Segment']
+            repr_string = 'Binary 4'
         elif representation == 'bin8':
-            header.append('Begin: Data Binary 8')
-            footer = ['End: Data Binary 8',
-                      'End: Segment']
+            repr_string = 'Binary 8'
         elif representation == 'txt':
-            header.append('Begin: Data Text')
-            footer = ['End: Data Text',
-                      'End: Segment']
+            repr_string = 'Text'
+        else:
+            raise ValueError(f'Unknown {representation=}.')
 
-        # Write header lines
-        with open(filename, 'w') as f:
-            f.write(''.join(map(lambda line: f'# {line}\n', header)))
+        header.append(f'Begin: Data {repr_string}')
+        footer = [f'End: Data {repr_string}',
+                  'End: Segment']
+
+        bheader = ''.join(f'# {line}\n' for line in header).encode('utf-8')
+        bfooter = ''.join(f'# {line}\n' for line in footer).encode('utf-8')
 
-        binary_reps = {'bin4': (1234567.0, 'f'),
-                       'bin8': (123456789012345.0, 'd')}
+        reordered = self.array.transpose((2, 1, 0, 3))  # ovf ordering
 
-        # Write data.
-        if representation in binary_reps:
-            # Reopen with binary write, appending to the end of the file.
-            with open(filename, 'ab') as f:
+        bin_rep = {'bin4': (1234567.0, '<f'),
+                   'bin8': (123456789012345.0, '<d')}
+
+        with open(filename, 'wb') as f:
+            f.write(bheader)
```

Comment:
`f.write(''.join(f'# {line}\n' for line in header).encode('utf-8'))`

Reply:
One variable less

Reply:
I've changed it differently. What do you think?

Reply:
Even better :)